Lib/smtplib.py

```diff
@@ -367,10 +367,10 @@ def send(self, s):
     def putcmd(self, cmd, args=""):
         """Send a command to the server."""
         if args == "":
-            str = '%s%s' % (cmd, CRLF)
+            str = cmd
         else:
-            str = '%s %s%s' % (cmd, args, CRLF)
-        self.send(str)
+            str = '%s %s' % (cmd, args)
+        self.send('%s%s' % (str.replace('\n', '\\n'), CRLF))
```

Comment:
Thanks for posting a PR.

A I posted in the issue, I think the correct fix would be to raise a ValueError if args contains \n *or* \r characters.  cmd should never be sourced from user input, so I don't think we need to check cmd, but as I said you could if you wanted to be paranoid :)

Note also that using 'str' as a variable name is not a good idea, since anyone reading python code is going to expect 'str' to be the type str.